### ISSUE TITLE
Build CUDA code with HIP

### ISSUE THREAD

**Issue by koparasy** [Olympus-HPC/proteus#75]
This is a low priority, but conceptually we should be able to handle HIP source code targeted to NVDA architectures. We should add tests for this.
